src/main/java/io/jenkins/plugins/gitlabserver/servers/GitLabServers.java

```diff
@@ -151,4 +158,24 @@ public void updateServer(@Nonnull GitLabServer endpoint) {
         }
         setServers(endpoints);
     }
+
+    /**
+     * Removes an endpoint.
+     * Checks if the GitLab Server name is matched
+     *
+     * @param name the server URL to remove.
+     * @return {@code true} if the list of endpoints was modified
+     */
+    public synchronized boolean removeServer(@CheckForNull String name) {
+        boolean modified = false;
+        List<GitLabServer> endpoints = new ArrayList<>(getServers());
```

Comment:
have you thought about using a HashTable to store the list of servers? That would make maintaining it simpler.

Reply:
Yeah, thought of it. But having an extra data structure for a few server connections (unless I see a use case where one user wants large number of server configurations) is an tradeoff where `space-complexity` might be more than `time-complexity`.

Reply:
This code is called so rarely, and the space is so small, that I don't think the trade-off matters. I'd go for simplicity of code

Reply:
In production code, most of the time the most expensive thing is time programmers spend maintaining it :-)

Reply:
Okay, I will implement it then. :+1:

Reply:
Any particular reason you're looking for the qualities of a `HashTable` over a `HashMap` here @jeffpearce?

Reply:
Just to simplify all of this code - you wouldn't need to have a loop to look up the entries you're interested in while maintaining the set of servers - you could just add and delete from it

Reply:
`HashTable` is synchronized while `HashMap` is not. I am wondering if guys have a suggestion.

Reply:
See **https://stackoverflow.com/questions/40471/differences-between-hashmap-and-hashtable**

Reply:
@justinharringa - sorry - I misread your question. I don't have a preference, but thought an associative data structure would be better than a list

Reply:
I would avoid hashtable/hashmap as they are not JCasC compatible (AFAIK, a JCasC test would confirm it) and I'd prefer that global configuration was configurable through configuration-as-code-plugin 😓

Reply:
I have limited knowledge of JCasC. If you have some time, can you implement a small part of what you want with JCasC support in this plugin?

Reply:
I want a simple test that confirms you can configure gitlabservers which is a global configuration and export.

You can configure any plugin in JCasC from a yaml file.

You should be intimate with the global configuration of the gitlab plugin, here you can see a demo for how to configure it in JCasC: https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos/gitlab

Reply:
Sure, I'll be happy to send a PR.

Though I don't see any CI on the project 😢
I'll add that too if you don't mind. We have a decent Travis file on JCasC: https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/.travis.yml

Reply:
That would be great. Thanks. :D

Reply:
Thanks for the info about JCasC @casz - definitely want to be compatible.